### ISSUE TITLE
broadcasting issue

### ISSUE THREAD

**Issue by FrankCAN** [PetarV-/GAT#20]
Hi Petar,
Many thanks for your code.

For the f_1 to transpose(f_2) operation, can I change the f_2 to f_2_neighbor (BxNxKx1) which means that obtain all the k neighbors first before broadcasting. And then operate f_1 + f_2_neighbor. The total dimension will become to BxNxKx1 directly. Do you think it is the same as sparse GAT? Many thanks for your help.

Best Regards
Frank

**Comment by light8lee** [comment]
Hello, do you figure it out this part? I'm also confused about the attn_head() function.

**Comment by xdql** [comment]
> Hello, do you figure it out this part? I'm also confused about the attn_head() function.

I also confused about the attn__head() function. can I learn more about you to figure it out.

**Comment by FrankCAN** [comment]
Hi,
Sorry for me late response, it is the same as sparse GAT.

**Comment by skx300** [comment]
I am trying to understand attn_head().  Here I have several questions:

`
       seq_fts = tf.layers.conv1d(seq, out_sz, 1, use_bias=False)

        # simplest self-attention possible
        f_1 = tf.layers.conv1d(seq_fts, 1, 1)
        f_2 = tf.layers.conv1d(seq_fts, 1, 1)
        # why the following is sum, why not multiplication? 
        logits = f_1 + tf.transpose(f_2, [0, 2, 1]) 
        coefs = tf.nn.softmax(tf.nn.leaky_relu(logits) + bias_mat)

        vals = tf.matmul(coefs, seq_fts)
        ret = tf.contrib.layers.bias_add(vals)
`

**Comment by FrankCAN** [comment]
Please check the paper Neural machine translation by jointly learning to align and translate.
